braindecode/datasets/base.py

```diff
@@ -9,55 +9,132 @@
-    info: pandas.DataFrame
+    windows: mne.Epochs
+        windows/supercrops obtained throiugh application of a windowing
+        function to a BaseDataset
+    description: pandas.DataFrame
         hold additional info about the windows
     """
-    def __init__(self, windows, info):
+    def __init__(self, windows, description):
         self.windows = windows
-        self.info = info
+        self.description = description
 
     def __getitem__(self, index):
-        target = self.windows.events[:,-1]
+        target = self.windows.metadata["target"]
         keys = ['i_supercrop_in_trial', 'i_start_in_trial', 'i_stop_in_trial']
-        info = self.windows.metadata.iloc[index][keys].to_list()
-        return self.windows[index].get_data().squeeze(0), target[index], info
+        supercrop_ind = self.windows.metadata.iloc[index][keys].to_list()
+        x = self.windows[index].get_data().squeeze(0)
+        return x, target[index], supercrop_ind
 
     def __len__(self):
-        return len(self.windows.events)
\ No newline at end of file
+        return len(self.windows.events)
+
+
+class BaseConcatDataset(ConcatDataset):
+    """A base class for concatenated datasets. Holds either mne.Raw or
+    mne.Epoch in self.datasets and has a pandas DataFrame with additional
+    description.
+
+    Parameters
+    ----------
+    list_of_ds: list
+        list of BaseDataset of WindowsDataset to be concatenated.
+    """
+    def __init__(self, list_of_ds):
+        super().__init__(list_of_ds)
+        self.description = pd.concat(ds.description for ds in list_of_ds)
+
+    def split(self, some_property=None, split_ids=None):
+        """Split the dataset based on some property listed in its description
+        DataFrame or based on indices.
+
+        Parameters
+        ----------
+        some_property: str
+            some property which is listed in info DataFrame
+        split_ids: list(int)
+            list of indices to be combined in a subset
+
+        Returns
+        -------
+        splits: dict{split_name: subset}
+            mapping of split name based on property or index based on split_ids
+            to subset of the data
+        """
+        assert split_ids is None or some_property is None, (
+            "can split either based on ids or based on some property")
+        if split_ids is None:
+            split_ids = _split_ids(self.description, some_property)
+        else:
+            split_ids = {split_i: split
+                         for split_i, split in enumerate(split_ids)}
+        # split_ids are indices for WindowsDatasets
+        supercrop_ids = _windows_dataset_ids_to_supercrop_ids(
+            split_ids, self.cumulative_sizes)
+        return {split_name: Subset(self, split)
+                for split_name, split in supercrop_ids.items()}
+
+
+def _windows_dataset_ids_to_supercrop_ids(dataset_ids, cumulative_sizes):
+    supercrop_ids = {}
+    for split_name, windows_is in dataset_ids.items():
```

Comment:
Not critical for now, but the `_is` prefix can be confusing - I suppose it's a pluralized version of `i`?